include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -631,6 +631,12 @@ namespace VectorTools
         if (!cell1->active() && !cell2->active())
           continue;
 
+        // skip foreign cells
+        if (cell1->active() && !cell1->is_locally_owned())
+          continue;
+        if (cell2->active() && !cell2->is_locally_owned())
+          continue;
```

Comment:
Well, it seems to work. :-)

No surprise though, because I require the corresponding cells to be owned by the same processor.